source/getting-started/connection.txt

```diff
@@ -0,0 +1,622 @@
+
+- ``DEFAULT``
+- ``SCRAM-SHA-256``
+- ``SCRAM-SHA-1``
+- ``MONGODB-CR``
+- ``X.509``
+- ``Kerberos (GSSAPI/SSPI)`` *(MongoDB Enterprise only)*
+- ``LDAP (PLAIN)`` *(MongoDB Enterprise only)*
+
+Connection Mechanisms
+---------------------
+
+In this section, you can find descriptions and sample code for connecting to
+a MongoDB server using each of the connection mechanism settings.
+
+DEFAULT
+~~~~~~~
+
+Specify the ``DEFAULT`` option to the driver in order to authenticate with the
+server using a username and password. This option is a fallback setting that
+instructs the driver to negotiate the first authentication mechanism supported
+by the server using the following order of preference:
+
+#. ``SCRAM-SHA-256``
+#. ``SCRAM-SHA-1``
+#. ``MONGODB-CR``
+
+If the ``DEFAULT`` option is specified, the driver first attempts to
+authenticate using ``SCRAM-SHA-256``. If the server does not support that
+mechanism, the driver attempts to authenticate using ``SCRAM-SHA-1``. If the
+server does not support that mechanism either, the driver attempts to
+authenticate using ``MONGODB-CR``.
+
+You can specify this option by omitting the ``authMechanism`` parameter of the
+`URI ConnectionString
+<https://docs.mongodb.com/manual/reference/connection-string/>`_ or by setting
+it explicitly to the value ``DEFAULT`` as shown in the following sample code.
+
+.. important::
+   Always **URI encode** the username and password using the
+   ``encodeURIComponent`` method to ensure they are correctly parsed.
+
+.. code-block:: js
+
+   const { MongoClient } = require("mongodb");
+
+   // specify the placeholder values for your environment in the following line
+   const username = encodeURIComponent("<username>");
+   const password = encodeURIComponent("<password>");
+   const clusterUrl = "<MongoDB cluster url>";
+   const authMechanism = "DEFAULT";
+
+   // Connection URI
+   const uri = `mongodb+srv://${username}:${password}@${clusterUrl}/?authMechanism=${authMechanism}`;
+
+   const client = new MongoClient(url);
+
+   async function run() {
+     try {
+       // Connect the client to the server
+       await client.connect();
+
+       // Establish and verify connection
+       await client.db("admin").command({ ping: 1 });
+       console.log("Connected successfully to server");
+     } finally {
+       // Ensures that the client will close when you finish/error
+       await client.close();
+     }
+   }
+   run().catch(console.dir);
+
+For more information on the challenge-response (CR) and salted
+challenge-response authentication mechanisms (SCRAM) that MongoDB supports,
+see the `SCRAM <https://docs.mongodb.com/manual/core/security-scram/>`_ section
+in the server manual.
+
+SCRAM-SHA-256
+~~~~~~~~~~~~~
+
+.. note::
+
+   ``SCRAM-SHA-256`` is the default authentication method for MongoDB starting
+   in version 4.0
+
+Specify the ``SCRAM-SHA-256`` option to the driver in order to authenticate
```

Comment:
Do you think it would be better to combine the first and second paragraph? Or, perhaps to leave out "specify X option" and instead describe what SCRAM-SHA-256 (or other connection mechanism) since what you're saying is restated in the second paragraph? E.g.

"The ``SCRAM-SHA-256`` authentication mechanism is a salted challenge-response mechanism (SCRAM) that uses your username and password, encrypted with the `SHA-256` algorithm to authenticate your user."

Reply:
I like this paragraph!